src/game/Spell.cpp

```diff
@@ -4264,7 +4264,10 @@ SpellCastResult Spell::CheckCast(bool strict)
             }
 
             if (!dispelTarget)
-                return SPELL_FAILED_NOTHING_TO_DISPEL;
+                if (m_spellInfo->Id == 24406) // Hunter's Improved Mend Pet
```

Comment:
Ah, I see. I'll test that in a bit.

The [Mend Pet](http://wow.gamepedia.com/Mend_Pet) spell in Cata up to WoD is no longer a channel (it is a channel again in Legions). [Improved Mend Pet](http://wow.gamepedia.com/Improved_Mend_Pet) itself was removed in MoP, but was altered in TBC to a 25%/50% chance.

I would guess a channel spell works differently compared to a HoT. I haven't looked.